### PR TITLE
Hackernews search in EVADB

### PR DESCRIPTION
**Pull request by kaushikravichandran**
This PR supports searching Hackernews stories in EVADB:

- We treat Hackernews as a data source
- Algolia search API for hackernews is used for searching
- Get requests are used
- We support searching within: `story`, `comment`, `URL`, `story`, `poll` or a combination of them
- Syntax:
```
params = {
        "query": "EVADB",
        "tags": "(story,poll)",
    }
query = f"""CREATE DATABASE hackernews_data
    WITH ENGINE = "hackernews",
    PARAMETERS = {params};"""
```

### CODE REVIEW

**Comment by xzdandy** [comment]
Thanks for adding Hackernews as data source? Could you also add a documentation page for it? 
https://evadb.readthedocs.io/en/stable/source/reference/databases/index.html

**Comment by kaushikravichandran** [comment]
> Thanks for adding Hackernews as data source? Could you also add a documentation page for it? https://evadb.readthedocs.io/en/stable/source/reference/databases/index.html

Added the docs. The docs test fail now: because I have added a link to the column info of the hackernews search results table: which is in this PR, and the link will become active once the PR is merged.
Link: https://github.com/georgia-tech-db/evadb/blob/staging/evadb/third_party/databases/hackernews/table_column_info.py

File: docs/source/reference/databases/hackernews.rst [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+* ``tags`` is the tag used for filtering the query results. Check `available tags <https://hn.algolia.com/api#:~:text=filter%20on%20a%20specific%20tag.%20Available%20tags%3A>`_ to see a list of available filter tags.
+
+Create Connection
+-----------------
+
+.. code-block:: text
+
+   CREATE DATABASE hackernews_data WITH ENGINE = 'hackernews', PARAMETERS = {
+        "owner": "EVADB",
+        "repo": "story"
```
Comment (xzdandy) [inline docs/source/reference/databases/hackernews.rst:30]: I think it should be query and tag instead of owner and repo.

Comment (kaushikravichandran) [reply docs/source/reference/databases/hackernews.rst]: Thanks! fixed it.

**Review by xzdandy — APPROVED** [review]
Looks great! Thanks.

**Pull request merged**